rmw_opensplice_cpp/src/functions.cpp

```diff
@@ -180,16 +180,23 @@ rmw_create_publisher(
     return nullptr;
   }
 
-  DDS::DataWriterQos default_datawriter_qos;
-  status = dds_publisher->get_default_datawriter_qos(default_datawriter_qos);
+  DDS::DataWriterQos datawriter_qos;
+  status = dds_publisher->get_default_datawriter_qos(datawriter_qos);
   if (status != DDS::RETCODE_OK) {
     rmw_set_error_string("failed to get default datawriter qos");
     // printf("get_default_datawriter_qos() failed. Status = %d\n", status);
     return nullptr;
   }
 
+  // ensure the history depth is at least the requested queue size
+  if (datawriter_qos.history.kind == DDS::KEEP_LAST_HISTORY_QOS &&
```

Comment:
When would the user have had the chance to do that? The publisher was just created in this function. At best they could do it after the publisher is returned from this function.